Check/Codebase/PhpCodeSniffer.php

```diff
@@ -0,0 +1,171 @@
+   */
+  public function getResultPass() {
+    return dt('No PHP code_sniffer violations.');
+  }
+
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultWarn().
+   */
+  public function getResultWarn() {
+    $ret_val = '';
+    if (drush_get_option('html') == TRUE) {
+      $ret_val .= '<table class="table table-condensed">';
+      $ret_val .= '<thead><tr><th>' . dt('Line, Column') . '</th><th>' . dt('Severity') . '</th><th>' . dt('Action') . '</th></tr></thead>';
+      foreach ($this->registry['phpcs_out'] as $filename => $violations) {
+        $num_violations = count($violations);
+        $ret_val .= "<tr align='center'><td colspan='3'><a href='file:///$filename'>File: $filename</a> Violations: $num_violations</td></tr>";
+        foreach ($violations as $violation) {
+          $line = $violation['line'];
+          $column = $violation['column'];
+          $severity = $violation['severity'];
+          $message = $violation['message'];
+          $ret_val .= "<tr><td>$line, $column</td><td>$severity</td><td>$message</td></tr>";
+        }
+      }
+      $ret_val .= '</table>';
+    }
+    else {
+      $rows = 0;
+      foreach ($this->registry['phpcs_out'] as $filename => $violations) {
+        if ($rows++ > 0) {
+          $ret_val .= PHP_EOL;
+          if (!drush_get_option('json')) {
+            $ret_val .= str_repeat(' ', 4);
+          }
+        }
+        $ret_val .= dt('Filename: @filename, Violations: @total', array(
+          '@filename' => $filename,
+          '@total' => count($violations),
+        ));
+        foreach ($violations as $violation) {
+          $ret_val .= PHP_EOL;
+          if (!drush_get_option('json')) {
+            $ret_val .= str_repeat(' ', 6);
+          }
+          $line = $violation['line'];
+          $column = $violation['column'];
+          $severity = $violation['severity'];
+          $message = $violation['message'];
+          $ret_val .= "$line, $column : $severity - $message";
+        }
+      }
+    }
+    return $ret_val;
+
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getAction().
+   */
+  public function getAction() {}
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\calculateScore().
+   */
+  public function calculateScore() {
+    // Get the path of phpcs.
+    $phpcs_path = $this->getExecPath('phpcs');
+    if ($phpcs_path === SiteAuditCheckAbstract::AUDIT_CHECK_SCORE_INFO) {
+      $this->registry['phpcs_path'] = $phpcs_path;
+      return $phpcs_path;
+    }
+    // Get the custom code paths.
+    $custom_code = $this->getCustomCodePaths();
+    if ($custom_code === SiteAuditCheckAbstract::AUDIT_CHECK_SCORE_INFO) {
+      $this->registry['custom_code'] = $custom_code;
+      return $custom_code;
+    }
+    // Get options.
+    // TODO: put in exhaustive list of ignore patterns.
+    $valid_options = array(
+      'extensions' => 'php,module,inc,install,test,profile,theme,css,info,txt',
+      'ignore' => '*/modules/features/',
```

Comment:
It's very common practice to put custom code in a feature's `.module` file; that's why I provided a comprehensive list of features files to ignore.